pages/apis/graphql_api.md.erb

```diff
@@ -55,6 +55,58 @@ curl https://graphql.buildkite.com/v1 \
 
 For documentation on the full list of fields and types, see the interactive documentation in the [GraphQL Explorer](#getting-started).
 
+## GraphQL IDs
+
+All node types have an `id` property, which is a global identifier encoding the object’s type and UUID. Buildkite’s GraphQL IDs are based on the [Relay Global Object ID](https://facebook.github.io/relay/graphql/objectidentification.htm).
```

Comment:
I've read through https://facebook.github.io/relay/graphql/objectidentification.htm twice now, and find it pretty tough going... largely because it's written as a specification for a server to comply with, rather than a documentation of the protocol a client can use (which are technically the same thing, but the former warrants a far more formal approach).

I'm also slightly unsure about how this implicitly subordinates our GraphQL API definition to Relay, which is a specific client library. :confused:

Reply:
Yeah, it's a really good point.

There's definitely people who will want to use a Relay client to talk to Buildkite. But that probably wasn't the audience for this section of the documentation… this was so we can link the "GraphQL ID" words on the [SSO Setup Guide](https://buildkite.com/docs/tutorials/sso-setup-with-graphql) to this new section.

Given that our ids are base64 encoded `{type}-{uuid}` Relay compatible ids, I wonder if we should still mention it though? Maybe not, and we just omit it entirely. We mention Relay in the section below anyhow.

@keithpitt @plasticine do GraphQL clients rely on being able to base64 decode the IDs, or are they meant to treat them as being completely opaque?

Reply:
> @keithpitt @plasticine do GraphQL clients rely on being able to base64 decode the IDs, or are they meant to treat them as being completely opaque?

They're supposed to just be rando strings, but they're unique to each object, and be used with the `node` query. Base64 encoding them is just the way we do it (and others/easiest). We could encrypt data into them and decrypt it server side and still be 100% compatible with the spec.

Reply:
Cool thanks! I'll go rework this a bit more then 👌🏼